src/platform/packages/shared/presentation/presentation_containers/interfaces/panel_management.ts

```diff
@@ -27,3 +29,25 @@ export interface CanExpandPanels {
 export const apiCanExpandPanels = (unknownApi: unknown | null): unknownApi is CanExpandPanels => {
   return Boolean((unknownApi as CanExpandPanels)?.expandPanel !== undefined);
 };
+
+// Disable panel management on certain panels with a control type
+// TypeScript lacks support for negated types as https://github.com/microsoft/TypeScript/pull/29317 was never merged
+// Therefore, we can't define types like IsDuplicable { type: string not ControlType }
+export interface IsNotDuplicable {
+  type: typeof ESQL_CONTROL;
+}
+
+export const apiCannotBeDuplicated = (unknownApi: unknown | null): unknownApi is IsNotDuplicable =>
+  apiIsOfType(unknownApi as IsNotDuplicable, ESQL_CONTROL);
+
+export interface IsNotExpandable {
+  type: ControlType;
+}
+export const apiCannotBeExpanded = (unknownApi: unknown | null): unknownApi is IsNotExpandable =>
+  apiIsOneOfType(unknownApi as IsNotDuplicable, CONTROL_TYPES);
```

Comment:
This kind of goes against the goal of the embeddable framework IMO. We shouldn't need to hardcode values like this - we should make all of our action compatibility decisions based on the contents of the API, not hardcoded values. 

A better approach, IMO, would be to add an `isExpandable` interface. We default this to `true` in `buildEmbeddable` if it is not provided - but `isExpandable` can be set to false in `initializeDataControl` (and separately for ES|QL controls).

This same logic could be used for the other interfaces you added here. What do you think?

Reply:
Makes sense, let me take a crack at it